Model/CustomerPaymentMethodsManagement.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Paytrail\PaymentService\Model;
+
+use Magento\Customer\Model\Session;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Vault\Api\PaymentTokenManagementInterface;
+use Psr\Log\LoggerInterface;
+
+class CustomerPaymentMethodsManagement
+{
+    /**
+     * @var Session
+     */
+    protected $customerSession;
+
+    /**
+     * @var PaymentTokenManagementInterface
+     */
+    protected $paymentTokenManagement;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    protected $searchCriteriaBuilder;
+
+    /**
+     * @var Json
+     */
+    protected $jsonSerializer;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @param Session $customerSession
+     * @param PaymentTokenManagementInterface $paymentTokenManagement
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param Json $jsonSerializer
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        Session                         $customerSession,
+        PaymentTokenManagementInterface $paymentTokenManagement,
+        SearchCriteriaBuilder           $searchCriteriaBuilder,
+        Json                            $jsonSerializer,
+        LoggerInterface                 $logger
+    ) {
+        $this->customerSession = $customerSession;
+        $this->paymentTokenManagement = $paymentTokenManagement;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->jsonSerializer = $jsonSerializer;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return array[]
```

Comment:
return type is not compliant with the interface. Implement the interface and then return the correct type.
Expected array of \Paytrail\PaymentService\Api\Data\CustomerPaymentMethodsResultInterface objects.
got multidimensional array